src/components/Capture/capture.js

```diff
@@ -60,23 +61,43 @@ class Capture extends Component {
     checkIfHasWebcam( hasWebcam => this.setState({hasWebcam}) )
   }
 
-  validateCapture = (id, valid) => {
-    const { actions, method } = this.props
-    actions.validateCapture({ id, valid, method})
+  maxAutomaticCaptures = 3
+
+  uploadCaptureToOnfido = () => {
+    this.setState({uploadInProgress: true})
+    const {validCaptures, method, side, token} = this.props
+    const capture = validCaptures[0]
+
+    postToOnfido(capture, method, token, this.state.advancedValidation,
+      (apiResponse) => this.confirmAndProceed(apiResponse, capture.id),
+      this.onApiError
+    )
   }
 
-  maxAutomaticCaptures = 3
+  confirmAndProceed = (apiResponse, id) => {
+    const {method, side, nextStep, actions: {confirmCapture}} = this.props
+    confirmCapture({method, id, onfidoId: apiResponse.id})
+    this.confirmEvent(method, side)
+    nextStep()
+  }
+
+  confirmEvent = (method, side) => {
+    if (method === 'document') {
+      if (side === 'front') events.emit('documentCapture')
+      else if (side === 'back') events.emit('documentBackCapture')
+    }
+    else if (method === 'face') events.emit('faceCapture')
+  }
 
   createCapture(payload) {
     const { actions, method, side } = this.props
     payload.side = side
     actions.createCapture({method, capture: payload, maxCaptures: this.maxAutomaticCaptures})
   }
 
-  onServerResponse = (response) => {
-    const { actions } = this.props
-    const valid = response.valid
-    this.validateCapture(response.id, valid)
+  onValidationServiceResponse = (id, {valid}) => {
```

Comment:
Just because Airbnb think it's a good idea doesn't mean it is! The reason they give is: `Why? Destructuring saves you from creating temporary references for those properties.` But what's wrong with creating a temporary reference?

We might create a small memory overhead, but never enough to affect performance. Whereas creating a temporary reference could make the code much more readable.